src/detection.jl

```diff
@@ -336,24 +336,23 @@ ColumnProperties(T) = ColumnProperties(T, 0x00)
     end
 end
 
+function findnextnewline(pos, stop, buf, opts)
+    while pos < stop
```

Comment:
`stop` here seems to be the last byte of the current chunk; why is it `<` here instead of `<=`? I understand the change below from `len = ranges[i + 1]` to `len = ranges[i + 1] - 1`, but it seems like we still want to check each byte in the chunk?

Reply:
(for example, on line 372 we're doing `while pos <= len`...)

Reply:
Thanks for the review!

This function `findnextnewline` only returns the position of the first encountered newline and defaults by returning `stop`. So, whether there is no newline, or whether the newline happens on the last byte, it will return `stop` either way and there is no side-effect to the function, so we might as well skip checking that last byte.
The default return value comes from the assumption that each chunk should end with a newline. Line 450 this assumption is verified by the preprocessing of ranges from lines 466-480, while on the call line 470 `stop` is simply the end of the buffer.